src/Timer.cc

```diff
@@ -20,11 +20,18 @@
 using namespace ignition;
 using namespace common;
 
+#ifndef _WIN32
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
 //////////////////////////////////////////////////
 Timer::Timer()
```

Comment:
I'm not so sure about deprecating this class, does chrono have a good substitute?